packages/openmrs/src/Adaptor.js

```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
  * Logs in to OpenMRS, gets a session token.
  * @example
  *  login(state)
- * @function
+ * @private
  * @param {State} state - Runtime state.
  * @returns {State}
  */
 function login(state) {
   const { instanceUrl, username, password } = state.configuration;
 
-  const params = {
-    method: 'GET',
-    url: `${instanceUrl}/ws/rest/v1/session`,
-    auth: { username, password },
-    jar: true,
-  };
-
+  const agent = request.agent();
   return new Promise((resolve, reject) => {
-    request(params, function (error, response, body) {
-      error = assembleError({ error, response, params });
-      if (error) {
-        reject(error);
-      } else {
-        const resp = tryJson(body);
-        resolve({ ...state, auth: resp });
-      }
-    });
+    agent
+      .get(`${instanceUrl}/ws/rest/v1/session`)
+      .auth(username, password)
+      .then(() => {
+        resolve({ ...state, agent });
+      });
   });
 }
 
 /**
  * Removes unserializable or confidential keys from the state.
  * @example
  *  cleanupState(state)
- * @function
+ * @private
  * @param {State} state
  * @returns {State}
  */
 function cleanupState(state) {
-  delete state.auth;
+  delete state.agent;
   return state;
 }
 
 /**
  * Gets patient matching a uuid
  * @example
- * execute(
- *   getPatient({ uuid: 123 })
- * )(state)
+ * getPatient("123")
  * @function
  * @param {object} params - object with uuid for the patient
  * @returns {Operation}
  */
-export function getPatient(params) {
+export function getPatient(uuid) {
   return state => {
-    const { uuid } = expandReferences(params)(state);
     console.log(`Searching for patient with uuid: ${uuid}`);
-
+    const { agent } = state;
     const { instanceUrl } = state.configuration;
 
     const url = `${instanceUrl}/ws/rest/v1/patient/${uuid}`;
 
-    return new Promise((resolve, reject) => {
-      request(
-        { url, method: 'GET', qs: { v: 'full' }, jar: true },
-        (error, response, body) => {
-          error = assembleError({ error, response });
-          if (error) {
-            reject(error);
-          } else {
-            console.log(`Success. Found patient.`);
-
-            const data = tryJson(body);
-            const nextState = composeNextState(state, data);
-            resolve(nextState);
-          }
+    return agent
+      .get(url)
+      .accept('json')
+      .query({ v: 'full' })
+      .end((error, response) => {
+        error = assembleError({ error, response });
+        if (error) {
+          throw error;
+        } else {
+          console.log(`Success. Found patient.`);
+
+          const data = tryJson(response.text);
+          const nextState = composeNextState(state, data);
+          return nextState;
```

Comment:
I see you're using `.end` here, and in other places. We need to pick one, either use the `.then` pattern or `.end`.